lib/engine/step/g_18_cz/token.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require_relative '../token'
+
+module Engine
+  module Step
+    module G18CZ
+      class Token < Token
+        def process_place_token(action)
+          ensure_ate_home_token_is_free(action)
+          super
+        end
+
+        def ensure_ate_home_token_is_free(action)
```

Comment:
seems like this method is not necessary, can move it in line

and also to clean it up

action.city.tile.cities.flat_map(&:tokens).any?

Reply:
Yeah i am a fan of more smaller functions but sure i will move it inline